vm/cuttlefish/cuttlefish.go

```diff
@@ -94,23 +104,38 @@ func (pool *Pool) Create(workdir string, index int) (vmimpl.Instance, error) {
 	return inst, nil
 }
 
-func (inst *instance) runOnHost(timeout time.Duration, cmd string) error {
-	outc, errc, err := inst.gceInst.Run(timeout, nil, cmd)
-	if err != nil {
-		return fmt.Errorf("failed to run command: %s", err)
-	}
-
-	for {
-		select {
-		case <-vmimpl.Shutdown:
-			return nil
-		case err := <-errc:
-			return err
-		case out, ok := <-outc:
-			if ok && inst.debug {
-				log.Logf(1, "%s", out)
-			}
+func (inst *instance) sshArgs(command string) []string {
+	sshArgs := append(vmimpl.SSHArgs(inst.debug, inst.sshKey, 22), inst.sshUser+"@"+inst.name)
+	if inst.sshUser != "root" {
+		return append(sshArgs, "sudo", "bash", "-c", "'"+command+"'")
+	}
+	return append(sshArgs, command)
+}
+
+func (inst *instance) runOnHost(timeout time.Duration, command string) error {
+	cmd := osutil.Command("ssh", inst.sshArgs(command)...)
+
+	var outBuf bytes.Buffer
+	cmd.Stdout = &outBuf
+	cmd.Stderr = &outBuf
+
+	if err := cmd.Start(); err != nil {
```

Comment:
I think this whole function can be replaced with osutil.RunCmd if we just need timeout.

But thinking of this more: instead of passing consoleReadCmd to Ctor we could add another exported function RunSomething which will accept consoleReadCmd to use. Then we could simply pass empty string here.

Reply:
I agree on the using osutil.RunCmd().

If we do add the RunSomething() function to gce.go, we'll need to export the instance type so that we can access the function here in cuttlefish.go. Is that what we want to do?